File: src/main/java/io/github/saneea/dvh/feature/file/WriteFile.java
package io.github.saneea.dvh.feature.file;

import java.io.BufferedOutputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import io.github.saneea.dvh.Feature;

public class WriteFile//
		extends FileContentTransfer//
		implements Feature.In.Bin.Stream {

	private InputStream in;

	@Override
	protected String getDescription() {
		return "transfer bytes from standard input to file";
	}

	@Override
	protected void handleFile(String fileName) throws IOException {
		try (OutputStream out = new BufferedOutputStream(//
				new FileOutputStream(fileName))) {
			in.transferTo(out);
		}
	}

	@Override
	public void setIn(InputStream in) {
		this.in = in;
	}

}

File: src/main/kotlin/io/github/saneea/dvh/feature/file/WriteFile.kt
package io.github.saneea.dvh.feature.file

import io.github.saneea.dvh.Feature
import java.io.File
import java.io.InputStream

class WriteFile :
    FileFeature(),
    Feature.In.Bin.Stream {

    override lateinit var inBinStream: InputStream

    override val description = "transfer bytes from standard input to file"

    override fun handleFile(file: File) {
        file
            .outputStream().buffered()
            .use(inBinStream::transferTo)
    }
}